### PR TITLE
Update types to be neutral about container runtime

### PR DESCRIPTION
**Pull request by sftim**
Replace references to Docker / Docker Engine with more neutral terminology. Also, replace one Docker Engine example with a containerd example.

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Replace some references to Docker / Docker Engine with more neutral terminology.

#### Which issue(s) this PR fixes:
(none)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

- KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2221-remove-dockershim/README.md

```docs

```

### CODE REVIEW

**Comment by liggitt** [comment]
lgtm, mind making the same changes in api/core/v1/types.go as well? (will require regenerating docs and proto files)

**Comment by sftim** [comment]
> making the same changes in `api/core/v1/types.go`

Ah, that's what I missed. I was expecting it to not go green and for that to be a hint about what to fix.

**Comment by sftim** [comment]
Relates to https://github.com/kubernetes/website/issues/30940

**Comment by sftim** [comment]
It looks like I'll need to build a VM / container that has the right flavor of Docker in it. I need a specific version of Docker that supports experimental features to be able to get the automated docs build to work.

There is an alternative: if someone can share a  tree / patch that's equivalent to the correct data for the 3rd (autogeneration) commit, I'd appreciate it.

**Comment by sftim** [comment]
This is ready for review :relieved:

**Comment by sftim** [comment]
/explain Why is Prow getting the work-in-progress label assbackwards?

**Comment by sftim** [comment]
Despite the labels, this is ready for review.

**Comment by sftim** [comment]
Trying something
/retitle [WIP] Update types to be neutral about container runtime

**Comment by sftim** [comment]
/retitle Update types to be neutral about container runtime

**Comment by sftim** [comment]
/priority important-soon
/milestone v1.24

Feel free to amend this. I'm aiming to get it merged and shipped for v1.24.

File: vendor/sigs.k8s.io/kustomize/kyaml/openapi/kubernetesapi/v1212/swagger.json [hunk 1]
```diff
@@ -5693,14 +5693,14 @@
...
+          "description": "Arguments to the entrypoint. The container image's CMD is used if this is not provided. Variable references $(VAR_NAME) are expanded using the container's environment. If a variable cannot be resolved, the reference in the input string will be unchanged. The $(VAR_NAME) syntax can be escaped with a double $$, ie: $$(VAR_NAME). Escaped references will never be expanded, regardless of whether the variable exists or not. Cannot be updated. More info: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell",
```
Comment (liggitt) [inline vendor/sigs.k8s.io/kustomize/kyaml/openapi/kubernetesapi/v1212/swagger.json:5696]: Content under vendor shouldn't be modified directly, this is a vendored snapshot

**Comment by liggitt** [comment]
one comment on the vendor change, lgtm otherwise

**Comment by dims** [comment]
@sftim one comment left to be addressed! PTAL

File: vendor/sigs.k8s.io/kustomize/kyaml/openapi/kubernetesapi/v1212/swagger.json (hunk 1 above)
Comment (sftim) [reply vendor/sigs.k8s.io/kustomize/kyaml/openapi/kubernetesapi/v1212/swagger.json:5696]: Fixed (reverted)

**Pull request merged**